tests/network/network_bonding_setup.pm

```diff
@@ -0,0 +1,311 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Setup simple network topology for bonding tests
+#
+# Maintainer: QE Core <qe-core@suse.de>
+#
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use power_action_utils "power_action";
+use serial_terminal 'select_serial_terminal';
+use network_utils qw(get_nics cidr_to_netmask is_nm_used is_wicked_used delete_all_existing_connections set_nics_link_speed_duplex check_connectivity_to_host_with_retry set_resolv set_nic_dhcp_auto reload_connections_until_all_ips_assigned);
+use Utils::Systemd qw(disable_and_stop_service systemctl check_unit_file);
+use lockapi;
+use utils;
+use version_utils;
+use registration qw(register_addons_cmd verify_scc investigate_log_empty_license);
+use transactional qw(trup_call process_reboot);
+
+my $server_ip = "10.0.2.101";
+my $subnet = "/24";
+my $gateway = "10.0.2.2";
+
+my $dhcpd_server_image_name = get_var "DHCPD_SERVER_IMAGE_NAME";
+my $dhcpd_server_image_registry = get_var "DHCPD_SERVER_IMAGE_REGISTRY";
+
+my $public_dns_string = get_var "PUBLIC_DNS", "8.8.8.8,8.8.4.4";
+my $scc_dns_string = get_var "SCC_DNS", "";
+my $registry_dns_string = get_var "REGISTRY_DNS", "";
+my $container_primary_dns_string = get_var "CONTAINER_PRIMARY_DNS", "8.8.8.8,8.8.4.4";
+my $override_dns_string = get_var "OVERRIDE_DNS", "";
+
+# Split the DNS strings into arrays only if the variable is defined and not empty
+my @public_dns = defined($public_dns_string) && $public_dns_string ne "" ? split(",", $public_dns_string) : ();
+my @scc_dns = defined($scc_dns_string) && $scc_dns_string ne "" ? split(",", $scc_dns_string) : ();
+my @registry_dns = defined($registry_dns_string) && $registry_dns_string ne "" ? split(",", $registry_dns_string) : ();
+my @container_primary_dns = defined($container_primary_dns_string) && $container_primary_dns_string ne "" ? split(",", $container_primary_dns_string) : ();
+
+my $requires_scc_registration = is_sle_micro || is_sle;
+
+sub register_scc {
+    my ($self) = @_;
+    my @cmd_parts = (
+        [''],    # Leading space for concatenation
+        ['-r', get_required_var('SCC_REGCODE')],    # Registration code
+    );
+
+    record_info("REGISTER_SCC");
+
+    # Conditionally add SCC_URL if it's set
+    if (my $scc_url = get_var('SCC_URL')) {
+        push @cmd_parts, ['--url', $scc_url];
+    }
+
+    my $scc_args = join ' ', map { join ' ', @$_ } @cmd_parts;
+
+    trup_call('register' . $scc_args) if is_transactional;
+    trup_call('--continue run zypper --gpg-auto-import-keys refresh') if is_transactional && is_staging;
+    script_output_retry("SUSEConnect $scc_args", retry => 5, delay => 60, timeout => 180) unless is_transactional;
+    process_reboot(trigger => 1, expected_grub => 1) if is_transactional;
+    select_serial_terminal;
+    zypper_call 'refresh';
+    zypper_call 'repos --details';
+}
```

Comment:
I think you can use this method instead, as it would handle both, transactional and non transactional systems properly (Current code, depends on having an already registered system for traditional sles)


https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/e676233ce421abe9bd4dd422fa2ae10c670bf267/lib/registration.pm#L178-L185

Reply:
I need to use SCC_URL, should I modify the code in [os-autoinst-distri-opensuse/lib/registration.pm](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/e676233ce421abe9bd4dd422fa2ae10c670bf267/lib/registration.pm#L178-L185)?

Reply:
@foursixnine what boorloader on micro ? https://openqa.suse.de/tests/15597759#step/disk_boot/3

Reply:
You can add also regurl here, but I;m not sure if that will work https://openqa.suse.de/tests/15589631#step/bootloader_uefi/3

Reply:
@foursixnine @dzedro Please check the new code :)

Reply:
https://openqa.suse.de/tests/15641318
https://openqa.suse.de/tests/15641440
https://openqa.suse.de/tests/15641442